### ISSUE TITLE
Failure to detect "Module has no attribute"

### ISSUE THREAD

**Issue by robin-wayve** [python/mypy#11184]
**Bug Report**

"error: Module has no attribute" is not caught in at least one context.

**To Reproduce**

Run `mypy` against the following examples with no command line flags or configuration.

**Expected Behavior**

Against this input as `test.py`...
```python
import os

os.islink('foo')  # wrong, should be os.path.islink
```
you correctly get an error.
```
test.py:3: error: Module has no attribute "islink"; maybe "link" or "symlink"?
Found 1 error in 1 file (checked 1 source file)
```

**Actual Behavior**
When you wrap the previous code in a function...
```python
import os

def this_is_fine():
  os.islink('foo')  # wrong, should be os.path.islink
```
it incorrectly passes the check
```
Success: no issues found in 1 source file
```

**Your Environment**

- Mypy version used: 0.910 (and 0.710)
- Mypy command-line flags: none
- Mypy configuration options from `mypy.ini` (and other config files): none
- Python version used: 3.6 (and 3.7)
- Operating system and version: Ubuntu Linux 18.04

**Comment by sobolevn** [comment]
Hi, @robin-wayve! This is a common thing with untyped functions. By default, `mypy` does not check them.

You have two options:
1. Use `--check-untyped-defs` option
2. Add type annotations to your function: `def this_is_fine() -> None:`

Docs: https://mypy.readthedocs.io/en/latest/common_issues.html#no-errors-reported-for-obviously-wrong-code

**Comment by robin-wayve** [comment]
Ah thanks! This makes sense: suspected I was indeed missing something.
